proto/documents/v1alpha/web_publishing.proto

```diff
@@ -0,0 +1,330 @@
+  // restricted in this API definition.
+  Member.Role role = 1;
+
+  // Optional. The timestamp after which the invite token will expire if not redeemed.
+  // If not provided, the server will decide the deadline based on the internal configuration.
+  google.protobuf.Timestamp expire_time = 2;
+}
+
+// Request to redeem an invite token. This is the most security-sensitive request,
+// because it allows to register new members on the site. The server must obtain
+// and verify the relation of the communicating peer with a corresponding Mintter Account.
+// After the token is redeemed, a new member on the site must be created, associating
+// the Mintter Account of the caller with the role invite token was created for.
+message RedeemInviteTokenRequest {
+  // Value of the invite token.
+  string token = 1;
+
+  // TODO: might not need these, because can be inferred from the communication channel.
+
+  // Mintter Account ID of the new member.
+  string account_id = 2;
+
+  // Mintter Device ID of the new member.
+  string device_id = 3;
+}
+
+// Response after the token is redeemed.
+message RedeemInviteTokenResponse {}
+
+// Request to get the site info.
+message GetSiteInfoRequest {}
+
+// Request to update site info. Doesn't support partial updates.
+message UpdateSiteInfoRequest {
+  // Title of the site.
+  string title = 1;
+
+  // Description of the site.
+  string description = 2;
+}
+
+// Request to list site members.
+message ListMembersRequest {
+  // Optional. Number of items per page.
+  int32 page_size = 1;
+
+  // Optional. Token for a specific page.
+  string page_token = 2;
+}
+
+// Response listing site members.
+message ListMembersResponse {
+  // List of members.
+  repeated Member members = 1;
+
+  // Optional token for the next page.
+  string next_page_token = 2;
+}
+
+// Request to get information about a specific member.
+message GetMemberRequest {
+  // Required. Mintter Account ID to get the information for.
+  string account_id = 1;
+}
+
+// Request to delete an existing member.
+message DeleteMemberRequest {
+  // Required. Mintter Account ID of the member to be deleted.
+  string account_id = 1;
+}
+
+// Request to publish a Mintter Document on a Mintter Web Site.
+message PublishDocumentRequest {
+  // Required. ID of the document to publish.
+  string document_id = 1;
+
+  // Required. Specific version of the document to publish.
+  string version = 2;
+
+  // Required. Path for the URL of this document on the web site.
+  string path = 3;
+
+  // List of other supporting materials that are necessary
+  // to properly render the document being published.
+  // Namely, originals that are transcluded (or possibly linked)
+  // in the document being published.
+  repeated ReferencedDocument referenced_documents = 4;
+}
+
+// Response when publishing a document.
+message PublishDocumentResponse {}
+
+// Request to unpublish a document from a site.
+message UnpublishDocumentRequest {
```

Comment:
Who should be able to Unpublish? @burdiyan @horacioh @ericvicenti 
- Site Owner only?
- Anyone with at least Editor permissions?
- The Author who initially published that document?

Reply:
Maybe it's safer to only let the owner do that. It would be weird if editors could unpublish articles from other editors, but letting them only unpublished their own articles may be a bit more cumbersome to implement right now.

Reply:
What I implemented was: You can unpublish if either (you are the site owner) or (you are the owner of the document and you have editor permissions) If someone revoked your permissions then you won't be able to unpublish the document from the site even if it's yours, is that ok, @burdiyan @ericvicenti @horacioh?